mlx/coverity_services.py

```diff
@@ -259,356 +209,237 @@ def add_filter_rqt(name, req_csv, valid_list, filter_list, allow_regex=False):
                         validated += delim + element
                         delim = ","
             else:
-                logging.error('Invalid %s filter: %s', name, field)
-        return validated
-
+                logging.error("Invalid %s filter: %s", name, field)
+        return validated, filter_list
 
-class CoverityDefectService(Service):
-    '''
-    Coverity Defect Service (WebServices)
-    '''
-
-    def __init__(self, config_service):
-        '''Create a Defect Service, bound to the given Configuration Service'''
-        super(CoverityDefectService, self).__init__(config_service.get_transport(),
-                                                    config_service.get_hostname(),
-                                                    config_service.get_port(),
-                                                    config_service.get_ws_version())
-        self.config_service = config_service
-        self.filters = ""
-        # logging.getLogger('suds.client').setLevel(logging.DEBUG)
-        url = self.get_ws_url('defectservice')
-        self.validate_presence(url, 'Coverity Defect Service')
+    def add_new_filters(self, request_filters, column_key, filter_list, matcher_type, matcher_class=None):
```

Comment:
I suggest removing the input argument `request_filters` and simply returning a list of new filters. This improves readability and testability.